File: linear_search.py

from utilities import generate_list, print_list

#Linear Search function
def lin_search(nums, key):
	for number in nums:
		if number == key:	return nums.index(number)
	return -1

if __name__ == '__main__':
	print "Enter Number of ints, range and key (betwenn 0 to ranege)  rexpectively:"
	n = int(raw_input())
	r = int(raw_input())
	key = int(raw_input())
	nums = generate_list(n,r)
	print_list(nums)
	index = lin_search(nums, key)
	if not index ==-1:	print "Key : ",key, " found at index : ",index
	else:				print "Key not found"

File: linear_search.c
/*
C program of Linear Search.
Input : Number of integers (n) range of numbers (r), key (key)
Output : Index of  key
*/
#include "c_utilities.h"
#include <stdlib.h>
#include <stdio.h>

//linear search function
int lin_search(int *arr, int n, int key);

int main(void)
{
	int n, r, key, index;
	int *arr;
	printf("Number of ints and Range resp. : \n");
	scanf("%d %d",&n, &r);
	printf("Key : \n");
	scanf("%d",&key);
	printf("Number of ints and Range resp. : %d and %d\n",n,r);
	arr = generate_array(n, r);
	print_array(arr, n);
	if ( (index = lin_search(arr, n, key)) != -1)
		printf("Index of key : %d is : %d \n",key,index);
	else
		printf("Key not found\n");
	free(arr);
	return 0;
}//main

int lin_search(int *arr, int n, int key)
{
	int i;
	for(i=0; i<n; i++)
	{
		if(key==arr[i])
			return i;
	}
	return -1;
}//lin_search
